### PR TITLE
✅ test: credential guard + branch guard hook integration tests (QA-002b-4)

### PR DESCRIPTION
**Pull request by Pushedskydiver**
## Summary

- **67 new integration tests** for credential guard and branch guard hooks via stdin/stdout JSON contract
- **Credential guard**: all 14 pattern categories (generic key/secret/token/password, AWS access+secret, GitHub PAT classic+fine-grained+OAuth, Slack xoxb+xoxp, Stripe live+test, 5 private key variants, Atlassian, Linear, 4 DB connection strings), all 5 allowed paths, edge cases (comments, test files, short strings, multiple credential types, Edit+MultiEdit, fail-open)
- **Branch guard**: force push (--force, -f, flag-after-branch, chained commands, --force-with-lease allowed), protected branches (main/master/develop, custom CLANCY_BASE_BRANCH, partial match allowed), destructive ops (reset --hard, clean -fd/-fdx, checkout -- ., restore ., branch -D), `git checkout .` without `--` documented as approved (regex requires `--`), allowed ops, disabled guard, fail-open
- Version bump to 0.8.14. Total: 1223 unit + 219 integration = 1442 tests

## Test plan

- [x] `npm test` — 1223 unit tests pass
- [x] `npm run test:integration` — 219 integration tests pass (67 new)
- [x] `npm run typecheck` — clean
- [x] `npm run lint` — clean
- [x] DA review — addressed: `git checkout .` behaviour documented, force-after-branch test, multiple credential types test

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -7,6 +7,14 @@ Headers: `✨ Features`, `🐛 Fixes`, `♻️ Refactors`, `✅ Tests`, `📝 Do
 
 ---
 
+## [0.8.14] — 2026-03-22
+
+### ✅ Tests
+
+- **Credential guard + branch guard hook integration tests (QA-002b-4)** — expanded hook scenario coverage via stdin/stdout JSON contract: all 14 credential pattern categories (generic key/secret/token/password, AWS access+secret, GitHub PAT classic+fine-grained+OAuth, Slack xoxb+xoxp, Stripe live+test, RSA/EC/DSA/OPENSSH private keys, Atlassian, Linear, MongoDB/PostgreSQL/MySQL/Redis connection strings), all 5 allowed paths, edge cases (comments, test files, short strings, fail-open), Edit+MultiEdit tool support. Branch guard: force push, protected branches (main/master/develop + custom CLANCY_BASE_BRANCH), destructive ops (reset --hard, clean -fd, checkout --, restore ., branch -D), allowed ops, disabled guard, fail-open. 1223 unit tests (unchanged) + 219 integration tests (67 new).
```
[the remark this answers is not in the document]
Comment (Pushedskydiver) [reply CHANGELOG.md]: Fixed — updated CHANGELOG to say argv JSON contract. See cce6844.
